apps/prairielearn/src/pages/authLogin/authLogin.ts

```diff
@@ -69,4 +72,37 @@ router.get(
   }),
 );
 
+const DevLoginParamsSchema = z.object({
+  uid: z.string().nonempty(),
+  name: z.string().nonempty(),
+  uin: z.string().nullable().optional().default(null),
+});
+
+router.post(
```

Comment:
I implemented this as a POST handler on this page instead of on `pages/devLogin` because generating a CSRF token for cross-URL POSTs is juuuuuust annoying enough that I didn't want to do it 😅